GRPC/grpcServer/grpcserver.go

```diff
@@ -33,42 +32,33 @@ func NewGRPCServer(service *service.Service) {
 	}
 
 }
-func (g *GRPCServer) GetUserWithRights(ctx context.Context, request *authProto.AccessToken) (*authProto.Response, error) {
-
-	return &authProto.Response{
-		UserId:      1,
-		Role:        1,
-		Permissions: request.AccessToken,
-	}, nil
-}
-func (g *GRPCServer) CheckToken(context.Context, *authProto.AccessToken) (*authProto.Result, error) {
-	return &authProto.Result{
-		Result: true,
-	}, nil
+func (g *GRPCServer) GetUserWithRights(ctx context.Context, request *authProto.AccessToken) (*authProto.UserRole, error) {
+	return g.service.Authorization.ParseToken(request.AccessToken)
 }
+
 func (g *GRPCServer) BindUserAndRole(ctx context.Context, user *authProto.User) (*authProto.ResultBinding, error) {
-	return &authProto.ResultBinding{}, g.service.AddRoleToUser(user)
+	return g.service.AddRoleToUser(user)
 }
 
-func (g *GRPCServer) TokenGenerationByRefresh(context.Context, *authProto.RefreshToken) (*authProto.GeneratedTokens, error) {
-	return nil, nil
+func (g *GRPCServer) TokenGenerationByRefresh(ctx context.Context, token *authProto.RefreshToken) (*authProto.GeneratedTokens, error) {
+	return g.service.RefreshTokens(token.RefreshToken)
 }
-func (g *GRPCServer) TokenGenerationById(ctx context.Context, user *authProto.User) (*authProto.GeneratedTokens, error) {
+func (g *GRPCServer) TokenGenerationByUserId(ctx context.Context, user *authProto.User) (*authProto.GeneratedTokens, error) {
 	if user.RoleId == 0 {
-		role, err := g.service.GetRoleById(int(user.UserId))
+		roleId, err := g.service.GetRoleByUserId(int(user.UserId))
 		if err != nil {
 			logger.Errorf("GetRoleById:%s", err)
 			return nil, fmt.Errorf("GetRoleById:%w", err)
 		}
-		user.RoleId = int32(role.ID)
+		user.RoleId = int32(roleId)
```

Comment:
if role should be integer why service returns it as string. If it is string, why you try to convert it to integer?

Reply:
Service returns roleId as int not as string, but GRPC functions do not know about such type. They accept int32 or int64.